### PR TITLE
Patch 1.18 fixes (somewhat)

### LINKED ISSUE #19: Latest patch/update causing error
Please see race department regarding the error that most of us are currently experiencing due to the latest change in F1 2020 files. Finding offsets does not work. It would be amazing if you could help the community by looking at this problem and fixing it. Thank you.

### PR DESCRIPTION
**Pull request by Electronics**
Patch 1.18 of the F1 2020 game has proven rather problematic... The memory address space used has apparently increased by 256x meaning my magic fix of `Find Offsets` a bit more non-solutioney. To trawl through all memory space used would take hours with the way I search for the various bits and bobs.

The fixes (or work-arounds) I have made:

- The default `Find Offsets` now considers **2**, larger, memory ranges. This now actually takes up to minutes to search through sadly.
- In some cases, this will still not find the correct starting addresses
- Added a new `Find Offsets with Custom Start` option in the `Game` menu. This allows for the search process to start at a specified address given in the popup in hexadecimal.

### tl;dr or how to make the F1NameChanger work now

If you want the quickest and most reliable method follow these steps:

1. Use [Cheat Engine](https://www.cheatengine.org/) to scan the F1 process for the `Value Type`: `string` of `{o:mixed}Carlos{/o}`.
2. There should only be one result, copy and paste the resulting address (`Ctrl+C` works) into popup searh offset of the new item `Game -> Find Offsets with Custom Start`
3. Should work as before!
4. This will likely need to be repeated every time the game starts :(

This should fix #19 #15 #18 #17

### CODE REVIEW

**Comment by werrer612** [comment]
is this a temporary fix or permanent?

**Comment by SergioGabi** [comment]
Thank you so much!!

**Comment by mg1519** [comment]
So has this been updated on RD? 

it would be good to update it on there and provide this information as well because there are still people making comments complaining on there 

They don't visit this site

**Comment by Electronics** [comment]
Ahh I'll do that now: definitely didn't forget!

**Comment by chenhsuan006** [comment]
thank you

**Comment by YOYIB** [comment]
i need video tutorial for this, I don't know how to make the Cheat Engine work, pleaseeee

**Comment by YOYIB** [comment]
@Electronics please, video tutorial

**Comment by werrer612** [comment]
yea even i dont know xd

**Comment by llucxdev** [comment]
The fix dodn't works for me, y really need another solution

**Comment by alamarca** [comment]
The fix dodn't works for me, y really need another solution

**Comment by alamarca** [comment]
the solution is very easy, within the F1 directory execute the F1_2020_dx12.exe and then do the process that is already detailed here

open the game F1_2020_dx12.exe and stay on the cover
open cheat engine
open F1 Change Name

in cheat engine, press the magnifying glass and select F1 2020
and now to the right in VALUE TYPE select STRING
and above in TEXT paste the following text {o:mixed}Carlos{/o}
press First Scan
will find a record we double click
now it is below we double click it
we copy the ADDRESS
let's go to F1 Change Name
we select Find offsets with custom start and paste the ADDRESS that we copied
at the end it will say INFO SUCCESS
now press Write to F1 and enjoy

I have the names of the pilots like this inside the file names.txt

Daniel RICCIARDO, Fernando ALONSO, ALO
Carlos SAINZ, Daniel RICCIARDO, RIC
Sebastian VETTEL, Carlos SAINZ, SAI
Sergio PEREZ, Sebastian VETTEL, VET
Alexander ALBON, Sergio PEREZ, PER
Daniil KVYAT, Yuki TSUNODA, TSU
Kevin MAGNUSSEN, Michael SCHUMACHER, SCH
Romain GROSJEAN, Nikita MAZEPIN, MAZ
Antonio GIOVINAZZI,
Charles LECLERC,
Esteban OCON,
George RUSSELL,
Kimi RÄIKKÖNEN,
Lance STROLL,
Lando NORRIS,
Lewis HAMILTON,
Max VERSTAPPEN,
Nicholas LATIFI,
Nico HÜLKENBERG,
Valtteri BOTTAS,

Alain FOREST,
Alain PROST,
Alessio LORANDI,
Alex MURRAY,
Anthoine HUBERT,
Antonio FUOCO,
Arjun MAINI,
Arron BARNES,
Artem MARKELOV,
Ayrton SENNA,
Benjamin COPPENS,
Brendon HARTLEY,
Callisto CALABRESI,
Callum ILOTT,
Christian LUNDGAARD,
DE VRIES,
Dan TICKTUM,
Daniel JONES,
Devon BUTLER,
Dorian BOCCOLACCI,
Esto SAARI,
Felipe DRUGOVICH,
Felipe MASSA,
Fernando ALONSO,
Flavio NIEVES,
Gert WALDMULLER,
Giuliano ALESI,
Guanyu ZHOU,
Guilherme SAMAIA,
Howard CLARKE,
Igor CORREIA,
Jack AITKEN,
Jack TREMBLAY,
Jay LETOURNEAU,
Jehan DARUVALA,
Jolyon PALMER,
Jonas SCHIFFER,
Jordan KING,
Juan Manuel CORREA,
Julian QUESADA,
Klimek MICHALSKI,
Louis DELÉTRAZ,
Luca GHIOTTO,
Lucas ROTH,
Lukas WEBER,
Mahaveer RAGHUNATHAN,
Marcus ARMSTRONG,
Marcus ERICSSON,
Marie LAURSEN,
Marino SATO,
Martin GILES,
Maximilian GÜNTHER,
Mick SCHUMACHER,
Naota IZUMI,
Nikita MAZEPIN,
Niko KARI,
Nirei FUKUZUMI,
Noah VISSER,
Nobuharu MATSUSHITA,
Pascal WEHRLEIN,
Pedro PIQUET,
Peter BELOUSOV,
Pierre GASLY,
Ralph BOSCHUNG,
Rashid NAIR,
Robert KUBICA,
Robert SHWARTZMAN,
Roberto MERHI,
Roy NISSANY,
Ruben MEIJER,
Santiago MORENO,
Sean GELAEL,
Sergey SIROTKIN,
Sophie LEVASSEUR,
Stoffel VANDOORNE,
Sérgio SETTE CÂMARA,
Tadasuke MAKINO,
Tatiana CALDERÓN,
Wilhelm KAUFMANN,
Yasar ATIYEH,
Yuki TSUNODA,
Michael SCHUMACHER,

==========================================================================================
abre el juego F1_2020_dx12.exe y quedate en la portada
abre cheat engine
abre F1 Change Name

en cheat engine, presiona la lupa y selecciona F1
y ahora a la derecha en VALUE TYPE selecciona STRING
y arriba en TEXT pega el siguiente texto {o:mixed}Carlos{/o}
presiona First Scan
encontrara un registro le hacemos doble clic
ahora se encuentra abajo le hacemos doble click 
copiamos el ADDRESS
vamos a F1 Change Name
seleccionamos Find offsets with custom start y pegamos el ADDRESS que copiamos
al final dira INFO SUCCESS
ahora presiona Write to F1 y disfruta

Los nombre de los pilotos los tengo asi dentro del archivo names.txt

Daniel RICCIARDO, Fernando ALONSO, ALO
Carlos SAINZ, Daniel RICCIARDO, RIC
Sebastian VETTEL, Carlos SAINZ, SAI
Sergio PEREZ, Sebastian VETTEL, VET
Alexander ALBON, Sergio PEREZ, PER
Daniil KVYAT, Yuki TSUNODA, TSU
Kevin MAGNUSSEN, Michael SCHUMACHER, SCH
Romain GROSJEAN, Nikita MAZEPIN, MAZ
Antonio GIOVINAZZI,
Charles LECLERC,
Esteban OCON,
George RUSSELL,
Kimi RÄIKKÖNEN,
Lance STROLL,
Lando NORRIS,
Lewis HAMILTON,
Max VERSTAPPEN,
Nicholas LATIFI,
Nico HÜLKENBERG,
Valtteri BOTTAS,

Alain FOREST,
Alain PROST,
Alessio LORANDI,
Alex MURRAY,
Anthoine HUBERT,
Antonio FUOCO,
Arjun MAINI,
Arron BARNES,
Artem MARKELOV,
Ayrton SENNA,
Benjamin COPPENS,
Brendon HARTLEY,
Callisto CALABRESI,
Callum ILOTT,
Christian LUNDGAARD,
DE VRIES,
Dan TICKTUM,
Daniel JONES,
Devon BUTLER,
Dorian BOCCOLACCI,
Esto SAARI,
Felipe DRUGOVICH,
Felipe MASSA,
Fernando ALONSO,
Flavio NIEVES,
Gert WALDMULLER,
Giuliano ALESI,
Guanyu ZHOU,
Guilherme SAMAIA,
Howard CLARKE,
Igor CORREIA,
Jack AITKEN,
Jack TREMBLAY,
Jay LETOURNEAU,
Jehan DARUVALA,
Jolyon PALMER,
Jonas SCHIFFER,
Jordan KING,
Juan Manuel CORREA,
Julian QUESADA,
Klimek MICHALSKI,
Louis DELÉTRAZ,
Luca GHIOTTO,
Lucas ROTH,
Lukas WEBER,
Mahaveer RAGHUNATHAN,
Marcus ARMSTRONG,
Marcus ERICSSON,
Marie LAURSEN,
Marino SATO,
Martin GILES,
Maximilian GÜNTHER,
Mick SCHUMACHER,
Naota IZUMI,
Nikita MAZEPIN,
Niko KARI,
Nirei FUKUZUMI,
Noah VISSER,
Nobuharu MATSUSHITA,
Pascal WEHRLEIN,
Pedro PIQUET,
Peter BELOUSOV,
Pierre GASLY,
Ralph BOSCHUNG,
Rashid NAIR,
Robert KUBICA,
Robert SHWARTZMAN,
Roberto MERHI,
Roy NISSANY,
Ruben MEIJER,
Santiago MORENO,
Sean GELAEL,
Sergey SIROTKIN,
Sophie LEVASSEUR,
Stoffel VANDOORNE,
Sérgio SETTE CÂMARA,
Tadasuke MAKINO,
Tatiana CALDERÓN,
Wilhelm KAUFMANN,
Yasar ATIYEH,
Yuki TSUNODA,
Michael SCHUMACHER,

=========================================================
abra o jogo F1_2020_dx12.exe e fique na capa
mecanismo de trapaça aberto
abrir F1 Alterar nome

no mecanismo de fraude, pressione a lupa e selecione F1 2020
e agora à direita em VALUE TYPE selecione STRING
e acima em TEXT cole o seguinte texto {o:mixed}Carlos{/o}
pressione Primeira digitalização
encontrará um registro que clicamos duas vezes
agora está abaixo, clicamos duas vezes nele
copiamos o ENDEREÇO
vamos para F1 Mudar nome
selecionamos Localizar deslocamentos com início personalizado e colamos o ENDEREÇO ​​que copiamos
no final dirá INFO SUCCESS
agora pressione Write to F1 e aproveite

Tenho os nomes dos pilotos assim dentro do arquivo names.txt

Daniel RICCIARDO, Fernando ALONSO, ALO
Carlos SAINZ, Daniel RICCIARDO, RIC
Sebastian VETTEL, Carlos SAINZ, SAI
Sergio PEREZ, Sebastian VETTEL, VET
Alexander ALBON, Sergio PEREZ, PER
Daniil KVYAT, Yuki TSUNODA, TSU
Kevin MAGNUSSEN, Michael SCHUMACHER, SCH
Romain GROSJEAN, Nikita MAZEPIN, MAZ
Antonio GIOVINAZZI,
Charles LECLERC,
Esteban OCON,
George RUSSELL,
Kimi RÄIKKÖNEN,
Lance STROLL,
Lando NORRIS,
Lewis HAMILTON,
Max VERSTAPPEN,
Nicholas LATIFI,
Nico HÜLKENBERG,
Valtteri BOTTAS,

Alain FOREST,
Alain PROST,
Alessio LORANDI,
Alex MURRAY,
Anthoine HUBERT,
Antonio FUOCO,
Arjun MAINI,
Arron BARNES,
Artem MARKELOV,
Ayrton SENNA,
Benjamin COPPENS,
Brendon HARTLEY,
Callisto CALABRESI,
Callum ILOTT,
Christian LUNDGAARD,
DE VRIES,
Dan TICKTUM,
Daniel JONES,
Devon BUTLER,
Dorian BOCCOLACCI,
Esto SAARI,
Felipe DRUGOVICH,
Felipe MASSA,
Fernando ALONSO,
Flavio NIEVES,
Gert WALDMULLER,
Giuliano ALESI,
Guanyu ZHOU,
Guilherme SAMAIA,
Howard CLARKE,
Igor CORREIA,
Jack AITKEN,
Jack TREMBLAY,
Jay LETOURNEAU,
Jehan DARUVALA,
Jolyon PALMER,
Jonas SCHIFFER,
Jordan KING,
Juan Manuel CORREA,
Julian QUESADA,
Klimek MICHALSKI,
Louis DELÉTRAZ,
Luca GHIOTTO,
Lucas ROTH,
Lukas WEBER,
Mahaveer RAGHUNATHAN,
Marcus ARMSTRONG,
Marcus ERICSSON,
Marie LAURSEN,
Marino SATO,
Martin GILES,
Maximilian GÜNTHER,
Mick SCHUMACHER,
Naota IZUMI,
Nikita MAZEPIN,
Niko KARI,
Nirei FUKUZUMI,
Noah VISSER,
Nobuharu MATSUSHITA,
Pascal WEHRLEIN,
Pedro PIQUET,
Peter BELOUSOV,
Pierre GASLY,
Ralph BOSCHUNG,
Rashid NAIR,
Robert KUBICA,
Robert SHWARTZMAN,
Roberto MERHI,
Roy NISSANY,
Ruben MEIJER,
Santiago MORENO,
Sean GELAEL,
Sergey SIROTKIN,
Sophie LEVASSEUR,
Stoffel VANDOORNE,
Sérgio SETTE CÂMARA,
Tadasuke MAKINO,
Tatiana CALDERÓN,
Wilhelm KAUFMANN,
Yasar ATIYEH,
Yuki TSUNODA,
Michael SCHUMACHER,
